packages/core/src/node/process/process.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, decorate } from 'inversify';
+import { ILogger } from '../../common/logger';
+import * as events from 'events';
+import * as child from 'child_process';
+import * as stream from 'stream';
+
+decorate(injectable(), events.EventEmitter);
+@injectable()
+export abstract class Process extends events.EventEmitter {
```

Comment:
Yes I like that delegation too and types too, I first started using node's events, then thinking our events would be nicer because of the reasons you expose I switched. But after discussing with @akosyakov I switched back to node events :)

I do think our events have some drawbacks, if you have multiple event types for example having 1 emitter per event type can be kinda verbose.

I also think node's eventemitter have a more feature rich API at the moment, we could get there too but for now...

So if it's OK with you I'll stick to node's events for now and we can talk about this in general with @akosyakov and the other in more detail ? 

I wish we could iron out when to use what.